website/files/models/osfstorage.py

```diff
@@ -82,9 +83,39 @@ def move_under(self, destination_parent, name=None):
         return super(OsfStorageFileNode, self).move_under(destination_parent, name)
 
     def save(self):
+        """
+        Saves node, logs any check-out or check-in actions
+            but we want to log specific activities (checkout, checkin)
+            check that via the if in the middle of the method and log the correct activity
+
+        :return:
+        """
         self.path = ''
         self.materialized_path = ''
-        return super(OsfStorageFileNode, self).save()
+        ret = super(OsfStorageFileNode, self).save()
+
+        if 'checkout' in ret and len(ret) == 1:
```

Comment:
Talked with @mfraezz about this. This came about because there were dreams last summer of having file metadata be generically modifiable by the API without doing model changes, thus allowing a new breed of apps in the OSF that may or may not overlap with things like JamDB based apps. The only thing we currently have that takes advantage of this is the file checkout functionality, and we may have decided to abandon that path. This issue makes that decision important.

There are essentially two major potential paths:

1. Make a generic "metadata changed" log event and have message like "Bob change the metadata of Fish Guy.stl from checkout: True to checkout: False." and keep the concept of API modifiable metadata without model changes;
2. Not have file metadata be so generically changeable and make a model method for each type of metadata that gets changed. Then make a specific log message for checking files in and out.